src/actions/issues.actions.js

```diff
@@ -131,3 +158,105 @@ export const setSortingInState = (sorting: string) => ({
 export const ResetIssuesSort = () => ({
   type: RESET_SORTING
 });
+
+export const handlePostComment = (
+  query: Object,
+  comment: Object,
+  token: string
+) =>
+  apiAction({
+    type: POST_COMMENT,
+    payload: {
+      path: `https://api.github.com/repos/${query.name}/${query.repo}/issues/${
+        query.number
+      }/comments?access_token=${token}`,
+      method: 'POST',
+      data: comment,
+      onSuccess: fetchIssueAfterComment
+    }
+  });
+
+export const addLabel = (query: Object, name) =>
+  apiAction({
+    type: ADD_LABEL,
+    payload: {
+      method: 'POST',
+      path: `https://api.github.com/repos/${query.name}/${query.repo}/issues/${
+        query.number
+      }/labels?access_token=${token}`,
+      data: { labels: [name] },
+      onSuccess: setLabels
+    }
+  });
+
+export const deleteLabel = (query: Object, labelName: string) =>
+  apiAction({
+    type: DELETE_LABEL,
+    payload: {
+      method: 'DELETE',
+      path: `https://api.github.com/repos/${query.name}/${query.repo}/issues/${
+        query.number
+      }/labels/${labelName}?access_token=${token}`,
+      onSuccess: setLabels
+    }
+  });
+
+export const setInitialLabels = ({ labels }) => ({
+  type: SET_LABELS,
+  payload: {
+    labels
+  }
+});
+
+export const setLabels = labels => ({
+  type: SET_LABELS,
+  payload: {
+    labels
```

Comment:
seems like you can use setLabels for set initial labels as well

Reply:
One of them gets a full issue and then destructure to get the labels, the other one (SET_LABELS) just get them right away.